src/lib/rosetta_lib/transaction.ml

```diff
@@ -223,6 +223,32 @@ module Signed = struct
       ; stake_delegation : Unsigned.Rendered.Delegation.t option
       }
     [@@deriving yojson]
+
+    let deprecated_fields =
+      [ "create_token"; "create_token_account"; "mint_tokens" ]
+
+    let of_yojson = function
+      | `Assoc l ->
+          let open Result.Let_syntax in
+          let%bind l =
+            let exception Non_null of string in
+            try
+              return
+              @@ List.filter l ~f:(fun (field, json) ->
+                     if List.mem ~equal:String.equal deprecated_fields field
+                     then
+                       match json with
+                       | `Null ->
+                           false
+                       | _ ->
+                           raise (Non_null field)
```

Comment:
**question**

Will this not cause the fold to raise early if there are multiple deprecated fields, but the first is populated.

For example wouldn't it raise here instead of filtering on the second field.
```json
{
  "signature": "a9ad136ccc8dfdfc76937dc884036d625cbdc6baadc968c13f2be090bf53a01f3e0cee3f0959c6a85f6d939ca08d9200ee15a0a64440a0f8e5affe8afde3251f",
  "payment": {
    "to": "B62qkcjubxpCvKyJvtisJqVrnNcXHE7EeZPbQ9FNJbTbFwXPVcYnjgX",
    "from": "B62qoigHEtJCoZ5ekbGHWyr9hYfc6fkZ2A41h9vvVZuvty9amzEz3yB",
    "fee": "10000000000",
    "token": "1",
    "nonce": "68",
    "memo": null,
    "amount": "16571009648348",
    "valid_until": "4294967295"
  },
  "stake_delegation": null,
  "create_token": <non null value>,
  "create_token_account": null,
  "mint_tokens": null
```

Reply:
Ah, good catch

Reply:
Yes, that's the intention. If there's a non-null value, the transaction is malformed and we want to abort with the error below.